tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -0,0 +1,226 @@
+	for _, secGroupRule := range secGroupRules {
+		if isNil(secGroupRule.SecurityRulePropertiesFormat) {
+			continue
+		}
+		secRulePropFormat = *(secGroupRule.SecurityRulePropertiesFormat)
+		if isNil(secRulePropFormat.DestinationPortRange) || isNil(secRulePropFormat.Priority) ||
+			isNil(secRulePropFormat.SourceAddressPrefix) {
+			continue
+		}
+		destPortRange = *(secRulePropFormat.DestinationPortRange)
+		protocol = secRulePropFormat.Protocol
+		access = secRulePropFormat.Access
+		priority = *(secRulePropFormat.Priority)
+		sourceAddressPrefix = *(secRulePropFormat.SourceAddressPrefix)
+		if destPortRange == winRM && access == winRmAction && protocol == winRMProtocol &&
+			priority == winRMPriority && len(sourceAddressPrefix) != 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// testWinRmPort checks if winRmHttps port is mentioned in the inbound security
+// group rules in the worker subnet.
+func testWinRmPort(t *testing.T) {
+	ctx := context.Background()
+	err := setup()
+	require.NoError(t, err, "failed to initialize azureProvider")
+	err = readInstallerInfo()
+	require.NoError(t, err, "failed to get info from wni file")
+	var secGroupPropFormat network.SecurityGroupPropertiesFormat
+	var secGroupRules []network.SecurityRule
+	for _, nsgName := range secGroupsIDs {
+		secGroupProfile, err := azureInfo.nsgClient.Get(ctx, azureInfo.resourceGroupName, nsgName, "")
+		assert.NoError(t, err, "failed to get the network security group profile")
+		assert.NotNil(t, secGroupProfile.SecurityGroupPropertiesFormat, "failed to get the security group properties format")
+		secGroupPropFormat = *(secGroupProfile.SecurityGroupPropertiesFormat)
+		assert.NotNil(t, secGroupProfile.SecurityRules, "failed to get the security rules list")
+		secGroupRules = *(secGroupPropFormat.SecurityRules)
+		assert.True(t, isWinRMPortOpen(secGroupRules), "winRmHttps port is not opened")
+	}
+}
+
+// createHostFile creates an ansible host file and returns the path of it
+func createHostFile(ip, password string) (string, error) {
+	hostFile, err := ioutil.TempFile("", "test")
+	if err != nil {
+		return "", fmt.Errorf("coud not make temporary file: %s", err)
+	}
+	defer hostFile.Close()
+
+	_, err = hostFile.WriteString(fmt.Sprintf(`[win]
+%s ansible_password=%s
+[win:vars]
+ansible_user=core
+ansible_port=%s
+ansible_connection=winrm
+ansible_winrm_server_cert_validation=ignore`, ip, password, winRM))
+	return hostFile.Name(), err
+}
+
+// testAnsiblePing checks if ansible is able to ping on opened winRmHttps port
+func testAnsiblePing(t *testing.T) {
+	for _, vmName := range instanceIDs {
+		vmRDPFilePath := dir + "/" + vmName
+		b, err := ioutil.ReadFile(vmRDPFilePath)
+		assert.NoError(t, err, "failed to read file %s", vmName)
+		// this regex looks for the IP address pattern from vmRDPFilePath.
+		// the sample vmRDPFilePath looks like xfreerdp /u:xxxx /v:xx.xx.xx.xx /h:1080 /w:1920 /p:'password1234'
+		ipAddressPattern := regexp.MustCompile(`\d+.\d+.\d+.\d+`)
```

Comment:
This is instance where you should define this outside the for loop as regex compile is expensive. 

Does this regex work if I have a password like "123.123.123.123"?